server/src/main/java/org/opensearch/search/SearchModule.java

```diff
@@ -335,6 +338,7 @@ public class SearchModule {
     private final SearchPlugin.ExecutorServiceProvider indexSearcherExecutorProvider;
 
     private final Collection<ConcurrentSearchRequestDecider.Factory> concurrentSearchDeciderFactories;
+    private final Map<Class<? extends Query>, Set<String>> profilerTimingsPerQuery = new HashMap<>();
```

Comment:
I'm not convinced that using Lucene `Query` classes as the keys for the custom timing is a good idea. In particular, there are a number of Lucene queries whose type changes as a result of the `rewrite` operation. In some cases, the rewritten query is an anonymous subclass.